### ISSUE TITLE
bug(datepicker): selected date range start and end focus is not visible in high contrast mode

### ISSUE THREAD

**Issue by dwaynelavon** [angular/components#23360]
## Steps to reproduce:

- Enable high contrast mode (Firefox)
- Click on date range picker toggle
- Select a start date and and end date
- Use keyboard to navigate through selected dates
- Notice that the start and end dates don't get an outline when focused

Expected Behavior
Start date and end date should get a focus indicator in HC mode

- StackBlitz: https://stackblitz.com/angular/xkadgxqyapm?file=src%2Fapp%2Fdate-range-picker-forms-example.ts

**Comment by zarend** [comment]
I cannot reproduce on the [Date range picker forms integration](https://material.angular.io/components/datepicker/examples#date-range-picker-forms) example. I believe this has already been fixed.

macos 12.5.1 (21G83)
Firefox version 104.0.2 (64-bit)
